### PR TITLE
[Windows] Artifact to create a URL shortcut

### PR DESCRIPTION
**Pull request by mithunshanbhag**
This artifact creates a .url (web) shortcut to specified internet/intranet resource. The shortcut file is created in the public desktop folder (%SystemDrive%\users\public\desktop). 

Parameters: 
- Shortcut Name: The name to assign to the shortcut file.
- Shortcut Target: The intranet or internet resource that'll be invoked by this shortcut.

Open Issues:
- Cannot use whitespaces or quotes in the parameter values.

### CODE REVIEW

File: Artifacts/windows-create-url-shortcut/Artifactfile.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    },
+    "ShortcutTargetPath": {
+      "type": "string",
+      "displayName": "Shortcut Target",
+      "description": "The intranet or internet resource that'll be invoked by this shortcut.",
+      "defaultValue": "http://portal.azure.com"
+    }
+  },
+  "runCommand": {
+    "commandToExecute": "[concat('powershell.exe -executionpolicy bypass -File CreateUrlShortcut.ps1', ' -ShortcutName ', replace(parameters('ShortcutName'), ' ', '_'), ' -ShortcutTargetPath ', parameters('ShortcutTargetPath'))]"
```
Comment (ryanvog) [inline Artifacts/windows-create-url-shortcut/Artifactfile.json]: What happens in the target path has a space in it?  Maybe consider replacing spaces in the file name too.

File: Artifacts/windows-create-url-shortcut/CreateUrlShortcut.ps1 [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+#  - Displays the script argument values (default or user-supplied).
+#
+# Parameters:
+#  - N/A.
+#
+# Return:
+#  - N/A.
+#
+# Notes:
+#  - Please ensure that the Initialize() method has been called at least once before this
```
Comment (ryanvog) [inline Artifacts/windows-create-url-shortcut/CreateUrlShortcut.ps1]: Can Initialize() be run idempotently?

**Comment by ryanvog** [comment]
LGTM

File: Artifacts/windows-create-url-shortcut/CreateUrlShortcut.ps1 (hunk 2 above)
Comment (mithunshanbhag) [inline Artifacts/windows-create-url-shortcut/CreateUrlShortcut.ps1]: No. Not designed to be idempotent. Just a simple private function (not a publicly exposed cmdlet).

File: Artifacts/windows-create-url-shortcut/Artifactfile.json (hunk 1 above)
Comment (mithunshanbhag) [inline Artifacts/windows-create-url-shortcut/Artifactfile.json]: The portion of the target path (URL) after the whitespace gets skipped. This is the bug/issue in the underlying Azure VM script extension infrastructure (that DTL artifacts rely on) that I was alluding to.

Comment (mithunshanbhag) [inline Artifacts/windows-create-url-shortcut/Artifactfile.json]: @ryanvog I dug a bit into this. Per the RFC spec, a URL is not allowed to have whitespaces. Whitespaces, if any, must be URL-encoded ('%20'). 

So I guess we got lucky :-). I'll modify the tooltip to explicitly mention what any user-supplied URLs must be URL-encoded.

http://stackoverflow.com/questions/497908/is-a-url-allowed-to-contain-a-space/497972#497972

**Pull request merged**